core/src/main/java/org/apache/calcite/sql/type/OperandTypes.java

```diff
@@ -986,6 +986,14 @@ public static SqlSingleOperandTypeChecker same(int operandCount,
 
   public static final SqlSingleOperandTypeChecker ANY_ANY =
       family(SqlTypeFamily.ANY, SqlTypeFamily.ANY);
+
+  public static final SqlSingleOperandTypeChecker SECOND_THIRD_SAME =
+      new SameOperandTypeChecker(3) {
```

Comment:
If you happened to see https://issues.apache.org/jira/browse/CALCITE-6383 you may have noticed that the SameOperatndTypeChecker does not enforce that the operands have the same type, but only that they are comparable. Perhaps if this is what you want you should document this fact here.

Reply:
I think I understand what you mean, like nvl2(2.0, 'abc', 2) is allowed in spark, but
nvl2(2.0, true, 2) is not allowed because Boolean and numeric are not comparable

Reply:
This is also the case with calcite currently. I have tested it.